### PR TITLE
Update push notification types, update notif navigation hook, and remove userInteraction check in onNotification method

### PR DESCRIPTION
**Pull request by Kyle-Shanks**
### Description
Lots of small updates to get push notification navigation working. The biggest change was the removal of the userInteraction check in the onNotification function. Also added more types and a notif type enum for push notifications

### Dragons

The removal of the userInteraction check could cause some unexpected behavior so I'll keep an eye out for that

### How Has This Been Tested?

Lots of manual testing, lots more to do

### How will this change be monitored?

N/A

### Feature Flags ###

N/A

### CODE REVIEW

File: packages/common/src/store/notifications/types.ts [hunk 1]
```diff
@@ -196,6 +234,70 @@ export type MilestoneNotification = BaseNotification &
       }
   )
 
+export type MilestoneFollowPushNotification = {
+  // TODO: Not the full structure. Need to verify the fields that come back from identity
```
Comment (sliptype) [inline packages/common/src/store/notifications/types.ts:238]: Do we still need all these todos?

File: packages/mobile/src/notifications.ts [hunk 2]
```diff
@@ -62,21 +62,21 @@ class PushNotifications {
...
-                body: notification.message.body ?? notification.message
-              }
-            : {})
-        })
-      )
-
-      this.navigation?.navigate(notification.data.data)
-    }
+    // NOTE: Removed the check for userInteraction
+    // It was not being set to true when the user pressed the push notification
```
Comment (sliptype) [inline packages/mobile/src/notifications.ts:66]: Does the app navigate when it's open and a new push notif comes in now? Seems like the behavior is different on ios and android

**Review by sliptype — APPROVED** [review]
Nice!

File: packages/mobile/src/notifications.ts (hunk 2 above)
Comment (Kyle-Shanks) [reply packages/mobile/src/notifications.ts:66]: Yea, I'll have to check this.

Comment (Kyle-Shanks) [reply packages/mobile/src/notifications.ts]: It was, compromise seems to be checking if the app is in the foreground or background when the notif comes in

File: packages/common/src/store/notifications/types.ts (hunk 1 above)
Comment (Kyle-Shanks) [reply packages/common/src/store/notifications/types.ts:238]: Yea they're helpful for people walking around this code to know what is complete vs half baked

File: packages/mobile/src/notifications.ts (hunk 2 above)
Comment (sliptype) [reply packages/mobile/src/notifications.ts]: Sounds like a plan!

**Pull request merged**